sklearn/tests/test_kernel_approximation.py

```diff
@@ -215,6 +215,33 @@ def test_rbf_sampler():
     assert np.mean(error) <= 0.05  # mean is fairly close
 
 
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_rbf_sampler_fitted_attributes_dtype(dtype):
+    # Test that attributes are float32 when input is float32
+    rbf = RBFSampler()
+
+    X = np.array([[1, 2], [3, 4], [5, 6]], dtype=dtype)
+
+    rbf.fit(X)
+
+    assert rbf.random_offset_.dtype == dtype
+    assert rbf.random_weights_.dtype == dtype
```

Comment:
You can use the ``global_dtype`` fixture.
```suggestion
def test_rbf_sampler_fitted_attributes_dtype(global_dtype):
    # Test that attributes are float32 when input is float32
    rbf = RBFSampler()

    X = np.array([[1, 2], [3, 4], [5, 6]], dtype=global_dtype)

    rbf.fit(X)

    assert rbf.random_offset_.dtype == global_dtype
    assert rbf.random_weights_.dtype == global_dtype
```

Reply:
When I make this change the `float32` test is skipped with the following message: Set SKLEARN_RUN_FLOAT32_TESTS=1 to run float32 dtype tests.

For a test like this it feels a bit weird to skip it by default no?

Reply:
We set this env var in 1 of the CI jobs, so it's tested in the CI. If you want to test locally, it's true that you have to set it as well. The goal of this fixture is to avoid having the whole test suite (there's an ongoing work to improve float32 coverage in tests) run twice everytime for every job. It's already long enough.